app/uk/gov/hmrc/tariffclassificationfrontend/controllers/AssignCaseController.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tariffclassificationfrontend.controllers
+
+import javax.inject.{Inject, Singleton}
+import play.api.i18n.MessagesApi
+import play.api.mvc._
+import uk.gov.hmrc.tariffclassificationfrontend.config.AppConfig
+import uk.gov.hmrc.tariffclassificationfrontend.models.Case
+import uk.gov.hmrc.tariffclassificationfrontend.models.request.AuthenticatedRequest
+import uk.gov.hmrc.tariffclassificationfrontend.service.CasesService
+import uk.gov.hmrc.tariffclassificationfrontend.views
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.Future.successful
+
+@Singleton
+class AssignCaseController @Inject()(authenticatedAction: AuthenticatedAction,
+                                     override val caseService: CasesService,
+                                     val messagesApi: MessagesApi,
+                                     override implicit val config: AppConfig) extends RenderCaseAction {
+
+
+  override protected def redirect: String => Call = routes.CaseController.trader
+
+  override protected def isValidCase(c: Case)(implicit request: AuthenticatedRequest[_]): Boolean = c.queueId.isDefined && !c.assignee.map(_.id).contains(request.operator.id)
+
+  def get(reference: String): Action[AnyContent] = authenticatedAction.async { implicit request =>
+    getCaseAndRenderView(reference, c => successful(views.html.assign_case(c)))
+  }
+
+  def post(reference: String): Action[AnyContent] = authenticatedAction.async { implicit request =>
+    def respond: Case => Future[Result] = {
```

Comment:
How if I want to assign the case to myself after it was assigned to a different user?
Your code does not allow it now

Reply:
It’s not a requirement at the moment (I double checked with Joe)

Reply:
yes, but we will need it soon I believe ...

Reply:
I suspect the UI will have a different pattern for that action - we can deal with it when we know